pruner/src/extended_unused.cpp

```diff
@@ -33,4 +58,69 @@ const IR::Node *PruneUnused::preorder(IR::Function *f) {
     return f;
 }
 
+// List unused struct declarations
+
+bool ListStructs::preorder(const IR::Member *p) {
+    auto *pexpr = (IR::PathExpression *)(p->expr);
```

Comment:
Don't do this. Instead use `p->expr->checkedTo<IR::PathExpression>()`. Very useful command. This will also fail if the expr is not a IR::PathExpression

Reply:
We actually want it to silently fail if it is not a path expression(for now atleast). I

Reply:
Well it depends, can this happen and what does it happen when it happens? We should always be programming defensively to avoid weird behavior. If it is not a pathexpression, this may be a nullptr or the behavior is undefined and then you crash. So we should guard against that. Either way, you should use To or checkedTo instead of these casts.

Reply:
So `to<IR::PathExpression>()` is a very common command, you will use it a lot when working with the compiler. It should work in this context and if it returns a `nullptr` you can return false or true. I believe you should return false, actually if I remember the visitor framework correctly.

Reply:
My bad, I must've been doing something wrong.

Reply:
I guess it depends on if you want to visit the children of this member, I am not if that is necessary.